utilities/travis-push-javadoc-to-gh-pages.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+#########################################################################################
+# This script generates javadocs for develop branch, moves them into gh-pages branch,
+# and pushes the changes to the paypal/SeLion repository on github.
+#
+# Generally timestamp is changed in the javadoc files each time we generate 
+# the files from mvn javadoc:jar 
+#
+# Commit is not performed if the changes are only related to timestamp.
+#########################################################################################
+
+################################
+# update the global git config 
+# user.name and user.email
+################################
+update_git_config() {
+    # update user email and name in git config for travis
+    git config --global user.email "travis@travis-ci.org"
+    git config --global user.name "travis-ci"
+}
+
+###############################
+# checkout the develop branch
+###############################
+switch_to_develop() {
+    git checkout develop
+    # verify that we are on develop branch, otherwise exit with error code
+    output=$(git rev-parse --abbrev-ref HEAD)
+    if [ "$output" = "develop" ]; then
+        echo "On develop branch"
+    else
+        echo "Failed to switch to develop branch"
+        exit 1
+    fi
+}
+
+##############################
+# checkout gh-pages branch
+##############################
+switch_to_ghpages() {
+    # add a writable remote that
+    git remote add -t gh-pages writable https://${GH_TOKEN}@github.com/paypal/SeLion
+    # get all the tracking branches 
+    git fetch -pq --all
+    # checkout the gh-pages branch
+    git checkout -q -b gh-pages -t writable/gh-pages
+    # verify that we are on gh-pages branch, otherwise exit with error code
+    output=$(git rev-parse --abbrev-ref HEAD)
+    if [ "$output" = "gh-pages" ]; then
+        echo "On gh-pages branch"
+    else
+        echo "Failed to switch to gh-pages branch"
+        exit 1
+    fi
+}
+
+##############################
+# replaces the current html 
+# apidocs on the 'gh-pages'
+# branch with content from 
+# the maven build
+##############################
+update_pages_from_output() {
+    # empty the folders html/java-docs/develop/{artifact}
+    rm -rf html/java-docs/develop/server-apis/*
+    rm -rf html/java-docs/develop/client-apis/*
+    rm -rf html/java-docs/develop/codegen-apis/*
+    rm -rf html/java-docs/develop/dataproviders-apis/*
+
+    # copy the javadoc files to respective folders from develop branch into gh-pages branch
+    cp -R client/target/apidocs/* html/java-docs/develop/client-apis/
+    cp -R codegen/target/apidocs/* html/java-docs/develop/codegen-apis/
+    cp -R server/target/apidocs/* html/java-docs/develop/server-apis/
+    cp -R dataproviders/target/apidocs/* html/java-docs/develop/dataproviders-apis/
```

Comment:
yes. there is an oversight here that dates back to when we created the common jar. I will fix it